src/wallet/wallet.cpp

```diff
@@ -4280,11 +4266,16 @@ bool DoMigration(CWallet& wallet, WalletContext& context, bilingual_str& error,
     }
 
     // Add the descriptors to wallet, remove LegacyScriptPubKeyMan, and cleanup txs and address book data
-    if (auto res_migration = wallet.ApplyMigrationData(*data); !res_migration) {
-        error = util::ErrorString(res_migration);
-        return false;
-    }
-    return true;
+    return RunWithinTxn(wallet.GetDatabase(), /*process_desc=*/"apply migration process", [&](WalletBatch& batch) EXCLUSIVE_LOCKS_REQUIRED(wallet.cs_wallet){
+        if (auto res_migration = wallet.ApplyMigrationData(batch, *data); !res_migration) {
+            error = util::ErrorString(res_migration);
+            return false;
+        }
+        // Process succeed, connect the SPKM signals
+        wallet.ConnectScriptPubKeyManNotifiers();
```

Comment:
In 868e2f64a9d7de3a980e64fd7acd7323214d3765 "wallet: ApplyMigrationData, group all db txs into a single atomic operation"

I don't think it's necessary to connect the signals anymore since the local wallet exists outside of any contexts. Although removing this should be its own commit.

If you choose to not remove this, for completeness, this should also do `NotifyCanGetAddressesChanged` to match the previous behavior.